src/averageAges.js

```diff
@@ -55,7 +75,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let filteredByChildren = people.filter(
+    x => people.some(c => c.name === x.mother));
```

Comment:
I'd rather create such a function (I'll need it later on)
```
const getMother = (mother) => people.find(
  ({ name }) => name === mother,
);
```
if `getMother` return `undefined`, it becomes `false`, otherwise – `true`
```suggestion
  let children = people.filter(({ mother }) => getMother(mother));
```